### PR TITLE
Removed rich embed filter as its no longer necessary.

### LINKED ISSUE #2977: Remove `rich_embed` filter
Discord seem to have fixed the bug that allowed user accounts to send embeds. Things that make me think this:
- I can't find a valid triggering of the filter in over 2 years
- This reddit post says they patched it https://www.reddit.com/r/Discord_selfbots/comments/sa0hc2/discord_embeds_patched/

This makes the rich embed filter no longer necessary. It's had quite a few bugs (mostly relating to twitter) and gives some false positives, so it would be helpful to have it removed.

### PR DESCRIPTION
**Pull request by Hashed0719**
closes #2977 .

### CODE REVIEW

**Review by wookie184 — APPROVED** [review]
LGTM, thanks.

I'll remove the filter on the server just before merging, once this has another approval from someone.

**Review by Xithrius — APPROVED** [review]
yep yep yep

**Comment by Xithrius** [comment]
> LGTM, thanks.
> 
> I'll remove the filter on the server just before merging, once this has another approval from someone.

I have removed the filter.

**Pull request merged**
